src/wgsl_reflect.js

```diff
@@ -379,12 +379,26 @@ export class WgslReflect {
             }
         }
 
-        const info = WgslReflect.TypeInfo[type.name];
-        if (info) {
-            return {
-                align: Math.max(explicitAlign, info.align),
-                size: Math.max(explicitSize, info.size)
-            };
+        {
+            const info = WgslReflect.TypeInfo[type.name];
+            if (info) {
+                const divisor = type.format === 'f16' ? 2 : 1

```

Comment:
TypeInfo has the alignment and size of "f16" that you defined below, set to 2 each. If you set the divisor to 2 for f16, alignment and size would be 1.

Reply:
No, because this code path only happens when it's `vec2<f16>` not when it's just `f16` (right?)

So in the `vec2<f16>` case it looks up `vec2` from the table which has f32/u32/u32 alignment/sizes. Checking the spec all the f16 sizes are half that.

Reply:
Sounds good.